src/processor/map/map_copy_from.cpp

```diff
@@ -133,26 +100,26 @@ std::unique_ptr<PhysicalOperator> PlanMapper::mapPartitioner(LogicalOperator* lo
         ku_dynamic_cast<LogicalOperator*, LogicalPartitioner*>(logicalOperator);
     auto prevOperator = mapOperator(logicalPartitioner->getChild(0).get());
     auto outFSchema = logicalPartitioner->getSchema();
+    auto& copyFromInfo = logicalPartitioner->copyFromInfo;
     std::vector<std::unique_ptr<PartitioningInfo>> infos;
     infos.reserve(logicalPartitioner->getNumInfos());
     for (auto i = 0u; i < logicalPartitioner->getNumInfos(); i++) {
-        auto info = logicalPartitioner->getInfo(i);
-        auto keyPos = getDataPos(*info->key, *outFSchema);
-        std::vector<std::string> columnNames;
-        std::vector<LogicalType> columnTypes;
-        getRelColumnNamesInCopyOrder(info->tableEntry, columnNames, columnTypes);
-        auto columnPositions = getColumnDataPositions(columnNames, info->payloads, *outFSchema);
-        infos.push_back(std::make_unique<PartitioningInfo>(keyPos, columnPositions,
-            std::move(columnTypes), PartitionerFunctions::partitionRelData));
+        infos.push_back(std::make_unique<PartitioningInfo>(logicalPartitioner->getInfo(i)->keyIdx,
+            PartitionerFunctions::partitionRelData));
     }
-    std::vector<common::logical_type_vec_t> columnTypes;
-    for (auto& info : infos) {
-        columnTypes.push_back(info->columnTypes);
-    }
-    auto sharedState = std::make_shared<PartitionerSharedState>(std::move(columnTypes));
+    std::vector<LogicalType> columnTypes;
+    std::vector<std::unique_ptr<evaluator::ExpressionEvaluator>> columnEvaluators;
+    getColumnEvaluators(copyFromInfo.columnExprs, outFSchema, columnEvaluators, columnTypes);
+    // Manually set columnTypes for _ID columns
+    columnTypes[0] = *LogicalType::INTERNAL_ID();
+    columnTypes[1] = *LogicalType::INTERNAL_ID();
+    columnTypes[2] = *LogicalType::INTERNAL_ID();
```

Comment:
I dont really like how this looks even though previously we did something similar. However, the alternative is to manually create copies of the INTERNAL_ID version of offsets in the binding stage similar to the pid one I did in rdf copy bind, which I also don't think looks clean either ...